cns/restserver/api.go

```diff
@@ -899,6 +899,109 @@ func (service *HTTPRestService) getNetworkContainerByOrchestratorContext(w http.
 	logger.Response(service.Name, getNetworkContainerResponse, returnCode, err)
 }
 
+// getAllNetworkContainers is used to check if NC refresh association is needed
+// If any NC from DNC request is not existing in CNS, refresh association will be triggered
+func (service *HTTPRestService) getAllNetworkContainers(w http.ResponseWriter, r *http.Request) {
+	logger.Printf("[Azure CNS] getAllNetworkContainers")
+
+	var req cns.GetAllNetworkContainersRequest
+	err := service.Listener.Decode(w, r, &req)
+	logger.Request(service.Name, &req, err)
+	if err != nil {
+		logger.Printf("[Azure CNS] getAllNetworkContainers failed with error: %s", err.Error())
+		return
+	}
+
+	var returnCode types.ResponseCode
+	var returnMessage string
+	var ncDetails map[string]cns.ContainerStatus
+	allNcExist := false
+
+	switch r.Method {
+	case "GET":
+		ncDetails, allNcExist = service.getAllNetworkContainersDetails(req.NetworkContainersID)
+
+	default:
+		returnMessage = "[Azure CNS] Error. getAllNetworkContainers did not receive a GET."
+		returnCode = types.InvalidParameter
+	}
```

Comment:
I have concerns about the dataflow here, it's pretty convoluted for how simple this request response is.
What if, early in the handler, we do a simple: 
```go
if r.Method != "GET" { return http.MethodNotAllowed } // but write the response 
``` 
and otherwise proceed with serving the request. Then there is no need to preallocate vars, maybe assign them in a switch, and return something later when it's not actually clear to the reader what has happened.
Plus, in the case where we do not handle the request, we should not return a half-populated response struct. We should *only* return an error.

Reply:
Addressed.